npc/re/jobs/2e/rebellion.txt

```diff
@@ -0,0 +1,1093 @@
+	mes("What time would you like to return?");
+	next;
+	.@select = select("Cancel", "Test monster spawn", "Start from beginning", "Furnace Zone", "Shooting Zone", "Complete the test");
+	if (.@select == 1) {
+		close;
+	} else if (.@select == 2) {
+		donpcevent("#target::OnStart");
+		donpcevent("#nontarget::OnStart");
+		close;
+	} else if (.@select == 4) {
+		.@questComplete = 12341;
+	} else if (.@select == 5) {
+		.@questComplete = 12342;
+	} else if (.@select == 6) {
+		.@questComplete = 12344;
+	}
+	erasequest(12340);
+	erasequest(12341);
+	erasequest(12342);
+	erasequest(12343);
+	erasequest(12344);
+	erasequest(12345);
+	for (.@i = 12340; .@i <= .@questComplete; .@i++) {
+		setquest(.@i);
+		completequest(.@i);
+	}
+	close;
+
+OnInit:
+	disablenpc(strnpcinfo(NPC_NAME));
+	end;
+}
+
+job_gun,56,26,0	script	rebelroom#1	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("einbroch", 49, 97);
+	end;
+}
+
+job_gun,18,10,0	script	rebelroom#2	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 81, 29);
+	end;
+}
+
+job_gun,165,22,0	script	rebelroom#3	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 196, 35);
+	end;
+}
+
+job_gun,197,49,0	script	rebelroom#4	WARPNPC,2,2,{
+	end;
+OnTouch:
+	warp("job_gun", 157, 21);
+	end;
+//--------- ??? ? ?? --------
+}
+
+job_gun,88,144,0	script	shootingrange#1	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 93, 123);
+	end;
+}
+
+job_gun,105,97,0	script	shootingrange#2	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 125, 98);
+	end;
+}
+
+job_gun,156,115,0	script	shootingrange#3	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 157, 137);
+	end;
+}
+
+job_gun,152,154,0	script	shootingrange#4	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 135, 154);
+	end;
+}
+
+job_gun,130,146,0	script	shootingrange#5	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 120, 130);
+	end;
+}
+
+job_gun,165,18,1	script	Furnace Controller#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != G_L_SHECIL) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	if (questprogress(12342) != 1) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	mes("Do you want to dredge the mold up from the furnace?");
+	next;
+	if (select("Quit.", "Begin.") == 1) {
+		mes("Switch to the Standby mode.");
+		close;
+	}
+	hideonnpc("Furnace Controller#1");
+	donpcevent("Furnace Controller#2::OnStart");
+	mes("Initializing Furnace Controller ZX-3100...");
+	close;
+
+OnInit:
+	hideoffnpc("Furnace Controller#1");
+	end;
+}
+
+job_gun,214,36,1	script	Auto Anvil#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (countitem(Tiny_Flame) > 0) {
+		mes("Unauthorized firearm creation has been detected. Report to Manager Elwin.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != 2431 || countitem(Steel_Article_) <= 4 || questprogress(12342) != 1) {
```

Comment:
```suggestion
	if (getstatus(SC_MONSTER_TRANSFORM, 1) != G_L_SHECIL || countitem(Steel_Article_) <= 4 || questprogress(12342) != 1) {
```

Reply:
done, will be pushed together in next commit